### PR TITLE
Fix RSS feed polling to handle unordered feeds and burst items

### PR DESCRIPTION
**Pull request by TheShield2594**
## Summary

This PR fixes several critical issues in RSS feed polling that caused feeds to go silent or post incomplete archives:

1. **Feed ordering**: The poller was taking the first item in the feed as "the latest," but RSS/Atom don't guarantee order. Feeds listed oldest-first would pin the cursor to an unchanging article and never post again.

2. **Burst handling**: When a feed published multiple items between sweeps, only the first was posted. Now up to 5 items per sweep are posted (newest kept), with the cursor advancing past the entire burst.

3. **Date parsing**: Unparseable `pubDate` values would throw `RangeError` when setting embed timestamps, aborting delivery before the cursor was written—causing the same feed to fail on every subsequent sweep.

4. **HTTP error handling**: Non-2xx responses were being passed to the parser instead of reporting the actual HTTP status, making it impossible to diagnose 403/404 errors.

5. **Poll boot recovery**: Polls with `endsAt: null` (no expiry) were being passed to `scheduleExpiry`, causing `getTime()` errors on every boot.

## Key Changes

- **RSS feed polling** (`rssService.js`):
  - Added `datedItems()` to extract and sort items by parsed date, filtering out unparseable dates
  - Added `MAX_ITEMS_PER_SWEEP` constant (5 items) to cap burst posting while advancing cursor past all new items
  - Refactored `deliverFeedUpdate()` to handle multiple items per feed, post oldest-first, and return count posted
  - First sight of a feed now posts only the newest item (not the whole archive)
  - Added sweep summary logging: feed count, items posted, failures, and parked feeds
  - Added daily news digest logging to distinguish "nothing new" from "all feeds unreachable"

- **Feed fetching** (`safeFeedFetch.js`):
  - Added `FEED_USER_AGENT` constant with proper bot identification and project URL
  - Added `Accept-Encoding: identity` header to prevent gzip responses that the parser can't decompress
  - Added HTTP status validation: non-2xx responses now throw with the status code instead of being parsed

- **Poll service** (`pollService.js`):
  - Fixed `scheduleActivePollExpirations()` to exclude `endsAt: null` polls (which have no expiry by design)

- **Tests**:
  - Refactored `rssXml()` helper to use new `rssXmlItems()` for flexible multi-item test feeds
  - Added comprehensive tests for feed ordering, burst handling, date parsing, and edge cases
  - Added tests for HTTP error reporting and feed fetch headers
  - Added tests for poll boot recovery with no-expiry polls

https://claude.ai/code/session_01GiDokzMyHcrWbmwapmyN7P

## Summary by CodeRabbit

- **New Features**
  - RSS updates can now deliver multiple new items per sweep, while preserving chronological order and limiting large bursts.
  - Feed processing now provides clearer status reporting for posted, skipped, failed, and unreachable feeds.

- **Bug Fixes**
  - Improved handling of unavailable or invalid RSS feeds with clearer HTTP errors.
  - Prevented startup errors for intentionally open polls without expiration times.
  - Improved compatibility with compressed feed responses.

### CODE REVIEW

File: src/services/rssService.js [hunk 1]
```diff
@@ -165,45 +165,81 @@ async function fetchSendableChannel(client, channelId) {
...
 
         // Targets the one subdocument rather than rewriting the whole
         // rssFeeds array, which is also what `guild.save()` on a
         // projected document could not do.
         await Guild.updateOne(
             { guildId: guild.guildId, 'rssFeeds._id': feed._id },
-            { $set: { 'rssFeeds.$.lastPublished': itemDate } }
+            { $set: { 'rssFeeds.$.lastPublished': newest } }
         );
+        return channel ? toPost.length : 0;
```
[the remark this answers is not in the document]
Comment (TheShield2594) [reply src/services/rssService.js]: Both halves confirmed and fixed in 6b4886d.

The partial-batch repeat was a regression this PR introduced: before batching, one item was posted per sweep or none was, so "sent" and "cursored" could not disagree. The cursor now tracks the last item whose `send` resolved and moves no further, so a batch that stops half way resumes exactly where it stopped. A completed batch still skips whatever `MAX_ITEMS_PER_SWEEP` left behind — that trade is deliberate and documented.

On the null channel: agreed, and it was pre-existing. `channels.fetch` failing with nothing in the cache is indistinguishable from a deleted channel, so advancing there dropped the burst permanently on a blip. It now writes nothing, at the cost of a no-op re-check each sweep while the channel is really gone — the cheaper of the two mistakes.

The cursor write also moved out of the delivery `try` into its own, so a failed write is reported as a failed write rather than as a delivery error.

Covered by two new cases in `tests/rssCheckFeeds.test.js` (partial batch cursors to the last item that landed; unreachable channel writes nothing). `tests/guildScanProjection.test.js` drove the old behaviour through a null channel — it asserts the *shape* of the targeted write, not that an undelivered item advances anything, so it now gets a channel that can send.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/services/rssService.js [hunk 2]
```diff
@@ -232,35 +268,45 @@ async function checkRssFeeds(client) {
         // workers — bounded parallelism without chunking (no worker idles
         // while a slow feed holds up its chunk).
         const urls = [...subscriptionsByUrl.keys()];
+        if (!urls.length) return;
```
[the remark this answers is not in the document]
Comment (TheShield2594) [reply src/services/rssService.js]: Fixed in 6b4886d — and you're right that this is exactly the gap the summary line exists to close, so leaving one case unlogged defeated it.

Rather than logging before the early return, the guard is gone: it was cosmetic, since `Promise.all` over an empty worker pool resolves and the `while (next < urls.length)` loop never runs. One branch less, and the sweep now always ends with its summary. Pinned by a new case asserting `[RSS] Sweep: 0 feed(s)` when no guild has a feed configured.

---
_Generated by [Claude Code](https://claude.ai/code)_
